tensorflow_probability/python/distributions/sigmoid_beta.py

```diff
@@ -0,0 +1,274 @@
+
+  Samples of this distribution are reparameterized (pathwise differentiable).
+  The derivatives are computed using the approach described in
+  [(Figurnov et al., 2018)][1].
+
+  #### Examples
+
+    ```python
+   tfd = tfp.distributions
+
+   dist = tfd.SigmoidBeta(concentration0=tf.constant(1),
+                          concentration1=tf.constant(2))
+
+   dist.sample([4, 5])  # Shape [4, 5, 3]
+
+   # `x` has three batch entries, each with two samples.
+   x = [[.1, .4, .5],
+        [.2, .3, .5]]
+   # Calculate the probability of each pair of samples under the corresponding
+   # distribution in `dist`.
+   dist.prob(x)         # Shape [2, 3]
+    ```
+
+  #### References
+
+  [1]: Michael Figurnov, Shakir Mohamed, Andriy Mnih.
+       Implicit Reparameterization Gradients. _arXiv preprint arXiv:1805.08498_,
+       2018. https://arxiv.org/abs/1805.08498
+
+  """
+  def __init__(self,
+               concentration1,
+               concentration0,
+               validate_args=False,
+               allow_nan_stats=True,
+               name='SigmoidBeta'):
+    """Initialize a batch of SigmoidBeta distributions.
+
+    Args:
+      concentration1: Positive floating-point `Tensor` indicating mean
+        number of successes; aka 'alpha'.
+      concentration0: Positive floating-point `Tensor` indicating mean
+        number of failures; aka 'beta'.
+      validate_args: Python `bool`, default `False`. When `True` distribution
+        parameters are checked for validity despite possibly degrading runtime
+        performance. When `False` invalid inputs may silently render incorrect
+        outputs.
+      allow_nan_stats: Python `bool`, default `True`. When `True`, statistics
+        (e.g., mean, mode, variance) use the value '`NaN`' to indicate the
+        result is undefined. When `False`, an exception is raised if one or
+        more of the statistic's batch members are undefined.
+      name: Python `str` name prefixed to Ops created by this class.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([concentration1, concentration0],
+                                       dtype_hint=tf.float32)
+      self._concentration1 = tensor_util.convert_nonref_to_tensor(
+          concentration1, dtype=dtype, name='concentration1')
+      self._concentration0 = tensor_util.convert_nonref_to_tensor(
+          concentration0, dtype=dtype, name='concentration0')
+
+    super(SigmoidBeta, self).__init__(
+        dtype=dtype,
+        validate_args=validate_args,
+        allow_nan_stats=allow_nan_stats,
+        reparameterization_type=reparameterization.FULLY_REPARAMETERIZED,
+        parameters=parameters,
+        name=name)
+
+  @classmethod
+  def _parameter_properties(cls, dtype, num_classes=None):
+    # pylint: disable=g-long-lambda
+    return dict(
+          concentration1=parameter_properties.ParameterProperties(
```

Comment:
I believe pylint should have problems with these lines?  Specifically this line should have 2 spaces removed (and continuations should have 4 spaces more than the previous lines).